apps/src/netsim/NetSimMessage.js

```diff
@@ -92,28 +92,72 @@ var NetSimMessage = module.exports = function (shard, messageRow) {
 NetSimMessage.inherits(NetSimEntity);
 
 /**
+ * @typedef {Object} MessageData
+ * @property {!number} fromNodeID - sender node ID
+ * @property {!number} toNodeID - destination node ID
+ * @property {!number} simulatedBy - node ID of client simulating message
+ * @property {*} payload - message content
+ * @property {number} extraHopsRemaining
+ * @property {number[]} visitedNodeIDs
+ */
+
+/**
+ * @typedef {Object} MessageRow
+ * @property {number} fromNodeID - this message in-flight-from node
+ * @property {number} toNodeID - this message in-flight-to node
+ * @property {number} simulatedBy - Node ID of the client responsible for
```

Comment:
I'd argue they should be nullable in `MessageData` but non-nullable in `MessageRow`, since `buildRowFromData` is using `utils.valueOr` to provide reasonable defaults.